hadoop-tools/hadoop-aws/src/test/java/org/apache/hadoop/fs/s3a/auth/ITestAssumeRole.java

```diff
@@ -152,6 +152,20 @@ public void testCreateCredentialProvider() throws IOException {
     }
   }
 
+  @Test
+  public void testCreateCredentialProviderWithExternalId() throws IOException {
```

Comment:
how about a test where a known invalid id set, with the expectation of a failure. that way wire up can be verified and a new stack trace for troubleshooting.md/assumed_roles.md is created.

use `intercept()` to catch the exception by type, but don't include any string match on the message text...that has turned out to be very brittle with SDK updates